lib/puppet/pops/types/p_sem_ver_range_type.rb

```diff
@@ -5,6 +5,10 @@ module Types
 #
 # @api public
 class PSemVerRangeType < PScalarType
+  def self.register_ptype(loader, ir)
+    create_ptype(loader, ir, 'ScalarType')
```

Comment:
Using a constant would not be ideal since the string becomes a type reference that is later resolved to the parent type. The parent type in turn, has a name, which is frozen. A frozen constant here would just result in two frozen strings with the same content (the name of the is constructed in the #create_ptype method and later frozen in the initializer, not reachable using a constant).